### PR TITLE
add U_GetString function

### PR DESCRIPTION
**Pull request by rfomin**
Revert '-' in TK_Identifier. U_GetString accepts all data except whitespace.

### CODE REVIEW

File: src/u_scanner.c [hunk 1]
```diff
@@ -216,6 +216,48 @@ static void U_RestoreState(u_scanner_t* s, u_scanner_t savedstate)
...
+  }
+
+  start = scanner->scanPos;
+  cur   = scanner->data[scanner->scanPos++];
+
+  while(scanner->scanPos < scanner->length)
+  {
+    cur = scanner->data[scanner->scanPos];
+
+    if(cur != ' ' && cur != '\t' && cur != 0 && (cur != '\n' || cur != '\r'))
```
Comment (fabiangreffrath) [inline src/u_scanner.c:249]: I don't understand that last condition. We want to call `break` if `cur` is either `'\n'` or `'\r'` (or `' '` or `'\t'` or `'\0'`), right? Maybe call `isgraph()` here for unambiguity? 😉

File: src/u_scanner.c [hunk 2]
```diff
@@ -250,7 +292,7 @@ boolean U_GetNextToken(u_scanner_t* scanner, boolean   expandState)
   cur   = scanner->data[scanner->scanPos++];
 
   // Determine by first character
-  if(cur == '_' || cur == '-' || (cur >= 'A' && cur <= 'Z') || (cur >= 'a' && cur <= 'z'))
+  if(cur == '_' || (cur >= 'A' && cur <= 'Z') || (cur >= 'a' && cur <= 'z'))
```
Comment (fabiangreffrath) [inline src/u_scanner.c:295]: This is already spec incompliant, as `TK_Identiier` tokens are expected to start with a letter.

Comment (rfomin) [reply src/u_scanner.c:295]: It's the same in PrBoom+: https://github.com/coelckers/prboom-plus/blob/969515162c5aebea4ad7a125ee178dcad3d576ad/prboom2/src/scanner.cpp#L252 I think we should keep it.

Comment (fabiangreffrath) [reply src/u_scanner.c:295]: Okay, sure.

File: src/u_scanner.c (hunk 1 above)
Comment (rfomin) [reply src/u_scanner.c:249]: Well in the same way we check for whitespace in `U_CheckForWhitespace`.

Comment (fabiangreffrath) [reply src/u_scanner.c:249]: This code is really confusing at times...

**Review by fabiangreffrath — APPROVED** [review]

File: src/u_scanner.c (hunk 1 above)
Comment (rfomin) [reply src/u_scanner.c:249]: Yes, I need to test it more. Maybe we should refactor the whitespace check.

Comment (fabiangreffrath) [reply src/u_scanner.c:249]: Well, I consider any string that contains control characters other than the "usual few" (`\t`, `\r`, `\n`) as malformed and demand any right to fail in one way or the other. 😁

**Pull request merged**